src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -162,28 +174,47 @@ class ActivityFeed extends React.Component<Props, State> {
         this.setState({ mentionSelectorContacts: getMentionWithQuery(searchStr) });
     };
 
+    componentDidMount(): void {
+        const { comments, tasks, versions } = this.props;
+        this.sortFeedItems(comments, tasks, versions);
+    }
+
+    componentWillReceiveProps(nextProps: any): void {
+        const { comments, tasks, versions } = nextProps;
+        this.sortFeedItems(comments, tasks, versions);
+    }
+
+    /**
+     * Sort feed valid feed items with new at the top, old at the bottom.
+     *
+     * @param args - Arguments list of each item container type that is allowed in the feed.
+     */
+    sortFeedItems(...args: Array<?Comments | ?Tasks | ?FileVersions>): void {
+        const feedItems = [];
+
+        // If all items are not ready, don't sort and render the feed
+        if (args.some((itemContainer) => !itemContainer || !itemContainer.entries)) {
```

Comment:
I think this can just be ```if(!comments || !task || !versions) return```

Reply:
I'd argue that this will allow us to add more feed types without having to modify this entire function.

Reply:
Yeah this is true, but even still the addition would be relatively minor. Would just have to add an argument, and 2 lines of code. Just thought the way I suggested was more readable, but if you disagree feel free to leave as is